fixham/internal/client/client.go

```diff
@@ -0,0 +1,56 @@
+package client
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/fixham/internal/docker"
+)
+
+type Client struct {
+	packageName   string
+	goBuildRunner *docker.GoBuildRunner
+}
+
+func NewClient(packageName string) *Client {
+	return &Client{
+		packageName:   packageName,
+		goBuildRunner: docker.NewGoBuildRunner(packageName),
+	}
+}
+
+func (c *Client) WithGoBuildVersion(version string) *Client {
+	c.goBuildRunner.WithVersion(version)
+	return c
+}
+
+func (c *Client) WithRoot(root string) *Client {
+	c.goBuildRunner.WithRepoVolume(root)
+	return c
+}
+
+func (c *Client) Lint(args ...string) {
+	gitConfigSSH := ""
+	if os.Getenv("GIT_USE_SSH") == "true" {
```

Comment:
This should be attached to a common Configuration struct rather than having `os.Getenv` embedded right in - make it harder to lookup what all the supported config options are this way. Then, `NewClient` would be updated like this:

```go
func NewClient(c config.Config) *Client { . . .}
```

And this line would be:

```go
if c.config.GitUseSSH {
}
```